src/main/java/nl/tudelft/ewi/devhub/server/database/controllers/PrivateRepositories.java

```diff
@@ -27,20 +27,20 @@ public PrivateRepository find(String username, String repositoryTitle) {
 		Preconditions.checkNotNull(username);
 		Preconditions.checkNotNull(repositoryTitle);
 
-		return ensureNotNull(query().from(privateRepository)
+		return ensureNotNull(query().selectFrom(privateRepository)
 				.where(privateRepository.owner.netId.equalsIgnoreCase(username)
 					.and(privateRepository.title.equalsIgnoreCase(repositoryTitle)))
-				.singleResult(privateRepository),
+				.fetchOne(),
```

Comment:
If fetchOne fails fast, then the ensureNotNull wrapper is unnecessary. But throughout Devhub we always propagate `EntityNotFound` exceptions, so if fetchOne throws something else, we should always catch and rethrow it, or may break other logic.